application/src/main/java/org/togetherjava/tjbot/commands/moderation/BanCommand.java

```diff
@@ -0,0 +1,181 @@
+    private static final String ACTION_VERB = "ban";
+    private final Predicate<String> hasRequiredRole;
+
+    /**
+     * Constructs an instance.
+     */
+    public BanCommand() {
+        super(COMMAND_NAME, "Bans the given user from the server", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, TARGET_OPTION, "The user who you want to ban", true)
+            .addOption(OptionType.STRING, REASON_OPTION, "Why the user should be banned", true)
+            .addOptions(new OptionData(OptionType.INTEGER, DELETE_HISTORY_OPTION,
+                    "the amount of days of the message history to delete, none means no messages are deleted.",
+                    true).addChoice("none", 0).addChoice("recent", 1).addChoice("all", 7));
+
+        hasRequiredRole = Pattern.compile(Config.getInstance().getHeavyModerationRolePattern())
+            .asMatchPredicate();
+    }
+
+    private static RestAction<InteractionHook> handleAlreadyBanned(@NotNull Guild.Ban ban,
+            @NotNull Interaction event) {
+        String reason = ban.getReason();
+        String reasonText =
+                reason == null || reason.isBlank() ? "" : " (reason: %s)".formatted(reason);
+
+        String message = "The user '%s' is already banned%s.".formatted(ban.getUser().getAsTag(),
+                reasonText);
+        return event.reply(message).setEphemeral(true);
+    }
+
+    @SuppressWarnings("MethodWithTooManyParameters")
+    private static RestAction<InteractionHook> banUser(@NotNull User target, @NotNull Member author,
+            @NotNull String reason, int deleteHistoryDays, @NotNull Guild guild,
+            @NotNull SlashCommandEvent event) {
+        return event.getJDA()
+            .openPrivateChannelById(target.getIdLong())
+            .flatMap(channel -> channel.sendMessage(
+                    """
+                            Hey there, sorry to tell you but unfortunately you have been banned from the server %s.
+                            If you think this was a mistake, please contact a moderator or admin of the server.
+                            The reason for the ban is: %s
+                            """
+                        .formatted(guild.getName(), reason)))
+            .mapToResult()
+            .flatMap(sendDmResult -> {
+                logger.info(
+                        "'{}' ({}) banned the user '{}' ({}) from guild '{}' and deleted their message history of the last {} days, for reason '{}'.",
+                        author.getUser().getAsTag(), author.getId(), target.getAsTag(),
+                        target.getId(), guild.getName(), deleteHistoryDays, reason);
+
+                return guild.ban(target, deleteHistoryDays, reason)
+                    .map(banResult -> sendDmResult.isSuccess());
+            })
+            .map(hasSentDm -> {
+                String dmNotice =
+                        Boolean.TRUE.equals(hasSentDm) ? "" : "(Unable to send them a DM.)";
```

Comment:
[See my comment on the kick, ban and unban command](https://github.com/Together-Java/TJ-Bot/pull/244#discussion_r748922925)

Reply:
done.